### ISSUE TITLE
"Implement Interface" Option Not Displayed in Right-Click Menu in .NET Class Library Projects

### ISSUE THREAD

**Issue by sebetci** [OmniSharp/omnisharp-vscode#4890]
## Issue Description ##

**"Implement Interface"** option not displayed in right-click menu in .NET Class Library projects.

## Steps to Reproduce ##

1. I'm starting new **.NET Class Library Project** using command line. I also manually create the **../Abstract** and **../Concrete** subdirectories.

```bash
dotnet new classlib -o ApplicationName.SubProjectName
cd ApplicationName.SubProjectName
mkdir Abstract
mkdir Concrete
```

2. I add an interface by right-clicking on the **Abstract** directory on VS Code and following `New C# > Interface`. I add a concrete class by right clicking on the **Concrete** directory and following the path `New C# > Class`.

```csharp
///<summary>Interface</summary>
using ApplicationName.SubProjectName.Entity;
namespace ApplicationName.SubProjectName.Abstract
{
   public interface IProductRepository
   {
      Product GetProductByID(int id);
   }
}

///<summary>Concrete Class</summary>
using ApplicationName.SubProjectName.Entity;
using ApplicationName.SubProjectName.Abstract;
namespace ApplicationName.SubProjectName.Concrete
{
   /// <summary>Shortcut "CTRL + ." doesn't work when cursor is on IProductRepository. </summary>
   public class MySQLProductRepository : IProductRepository{ /* ... */}
}
```

3.  When I use the "**CTRL + .**" shortcut while on the interface, the **"Implement Interface"** option does not appear in the right-click menu. When I compile the project for testing, I get the error that the interface is not implemented. When I implement the interface manually, the project compiles successfully. 

```bash
dotnet build
```

This issue doesn't happen when using other Microsoft Technologies such as **ASP.NET Core**, **ADO.NET**, etc. But whenever I have a project where I'm creating a **.NET Class Library** this problem occurs.

![image](https://user-images.githubusercontent.com/86542016/141522679-87f38a82-8b5d-43b4-9141-9496d6d6b05c.png)

## Expected Behavior ##

When the **"CTRL + ."** shortcut is used while the cursor is on the interface name, the interface must be implemented on the concrete class.

## Actual Behavior ##

When using the **"CTRL + ."** shortcut while the cursor is on the interface name, the "Implement Interface" option is not displayed in the right-click menu.

## Environment information ##

**VSCode version**: 1.62.2
**C# Extension**: 1.23.16

<details><summary>Dotnet Information</summary>
.NET SDK (reflecting any global.json):
 Version:   5.0.403
 Commit:    bf9d168ac2

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19042
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.403\

Host (useful for support):
  Version: 5.0.12
  Commit:  7211aa01b3

.NET SDKs installed:
  3.1.415 [C:\Program Files\dotnet\sdk]
  5.0.209 [C:\Program Files\dotnet\sdk]
  5.0.402 [C:\Program Files\dotnet\sdk]
  5.0.403 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.21 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.20 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.21 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.20 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.21 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|
|---|---|---|
|add-reference|adrianwilczynski|1.0.2|
|angular-bootstrap|herrherrmann|4.0.9|
|angular-console|nrwl|17.12.2|
|angular-essentials|johnpapa|12.0.0|
|angular-schematics|cyrilletuzi|5.0.1|
|Angular2|johnpapa|12.0.0|
|asp-net-core-switcher|adrianwilczynski|2.0.2|
|beautify|HookyQR|1.5.0|
|better-comments|aaron-bond|2.1.0|
|blazor-snippet-pack|adrianwilczynski|2.6.1|
|bootstrap-4-cdn-snippet|eventyret|1.13.0|
|bootstrap-5-snippets|babobski|0.1.1|
|bootstrap-5-snippets-by-coder-foundry|CoderFoundry|1.2.1|
|bootstrap-v4-snippets|Zaczero|1.1.3|
|bootstrap4-vscode|thekalinga|6.1.0|
|bootstrap5-snippets|HansUXdev|1.2.5|
|bootstrap5-vscode|AnbuselvanRocky|0.2.0|
|code-stats-vscode|riussi|1.0.18|
|codesnap|adpyke|1.3.4|
|cpptools|ms-vscode|1.7.1|
|csharp|ms-dotnettools|1.23.16|
|csharp-to-typescript|adrianwilczynski|1.12.1|
|csharpextensions|kreativ-software|1.5.2|
|debugger-for-chrome|msjsdiag|4.13.0|
|debugger-for-edge|msjsdiag|1.0.15|
|docomment|k--kato|0.1.30|
|dotnet|formulahendry|0.0.4|
|dotnet-test-explorer|formulahendry|0.7.7|
|EditorConfig|EditorConfig|0.16.4|
|git-extension-pack|donjayamanne|0.1.3|
|git-graph|mhutchie|1.30.0|
|git-project-manager|felipecaputo|1.8.1|
|gitblame|waderyan|8.1.0|
|githd|huizhou|2.2.4|
|githistory|donjayamanne|0.6.19|
|github-markdown-preview|bierner|0.1.0|
|github-vscode-theme|GitHub|5.0.3|
|gitignore|codezombiech|0.7.0|
|gitlab-workflow|GitLab|3.37.0|
|gitlens|eamodio|11.6.1|
|JavaScriptSnippets|xabikos|1.8.0|
|libman|adrianwilczynski|1.0.1|
|LiveServer|ritwickdey|5.6.1|
|markdown|starkwang|1.2.5|
|markdown-checkbox|bierner|0.3.2|
|markdown-emoji|bierner|0.2.1|
|markdown-image|hancel|1.1.14|
|markdown-preview-enhanced|shd101wyy|0.6.1|
|markdown-preview-github-styles|bierner|0.2.0|
|markdown-shortcuts|mdickin|0.12.0|
|markdown-table-prettify|darkriszty|3.6.0|
|markdown-toc|AlanWalk|1.5.6|
|markdown-yaml-preamble|bierner|0.1.0|
|material-icon-theme|PKief|4.10.0|
|mermaid-markdown-syntax-highlighting|bpruitt-goddard|1.2.2|
|mithril-emmet|FallenMax|0.7.6|
|mongodb-vscode|mongodb|0.6.14|
|msbuild-project-tools|tintoy|0.3.15|
|namespace|adrianwilczynski|1.1.2|
|netcore-extension-pack|doggy8088|1.5.2|
|netcore-snippets|doggy8088|3.15.2|
|ng-template|Angular|13.0.0|
|path-intellisense|christian-kohler|2.4.2|
|polacode|pnp|0.3.4|
|prettier-vscode|esbenp|9.0.0|
|project-manager|alefragnani|12.4.0|
|quicktype-refresh|doggy8088|0.1.1|
|regex|chrmarti|0.4.0|
|remote-wsl|ms-vscode-remote|0.58.5|
|seti-icons|qinjia|0.1.3|
|start-git-bash|McCarter|1.2.1|
|terminal|formulahendry|0.0.10|
|terminal-commands|adrianwilczynski|1.0.5|
|theme-dracula|dracula-theme|2.24.0|
|toggle-hidden|adrianwilczynski|1.0.2|
|user-secrets|adrianwilczynski|2.0.1|
|vscode-angular2-emmet|jakethashi|2.0.3|
|vscode-angular2-files|alexiv|1.6.4|
|vscode-eslint|dbaeumer|2.2.2|
|vscode-github|KnisterPeter|0.30.7|
|vscode-language-pack-tr|MS-CEINTL|1.62.3|
|vscode-npm-script|eg2|0.3.23|
|vscode-nuget-package-manager|jmrog|1.1.6|
|vscode-open-in-github|ziyasal|1.3.6|
|vscode-peacock|johnpapa|3.10.1|
|vscode-proto3|zxh404|0.5.5|
|vscode-pull-request-github|GitHub|0.32.0|
|vscode-terminal-here|Tyriar|0.2.4|
|vscode-todo-highlight|wayou|1.0.5|
|vscode-versionlens|pflannery|1.0.9|
|winteriscoming|johnpapa|1.4.4|;

</details>

**Comment by sebetci** [comment]
I just realized that this issue is because OmniSharp stopped working on Visual Studio Code for some reason I don't know.

Therefore, I solved the problem by following the steps below on Visual Studio Code:

1. Open the command palette using the **CTRL + SHIFT + P** shortcut.
2. Search for **"OmniSharp: Select Project"**.
3. Select the project you are working on.

When I enabled OmniSharp on existing project by following these steps the problem was resolved. The directory I opened from the Visual Studio Code consists of five subprojects and dozens of source code files; this is the first time I have had this problem. I think this problem is caused by the large amount of files/directories tracked by OmniSharp. That's why after a point C# intellisense stops working.
